src/aind_metadata_upgrader/procedures_upgrade.py

```diff
@@ -0,0 +1,459 @@
+            "craniotomy_hemisphere": get_or_default(old_subj_procedure, Craniotomy, "craniotomy_hemisphere"),
+            "bregma_to_lambda_distance": old_subj_procedure.get("bregma_to_lambda_distance", None),
+            "bregma_to_lambda_unit": get_or_default(old_subj_procedure, Craniotomy, "bregma_to_lambda_unit"),
+            "implant_part_number": old_subj_procedure.get("implant_part_number", None),
+            "dura_removed": get_or_default(old_subj_procedure, Craniotomy, "dura_removed"),
+            "protective_material": get_or_default(old_subj_procedure, Craniotomy, "protective_material"),
+            "recovery_time": old_subj_procedure.get("recovery_time", None),
+            "recovery_time_unit": get_or_default(old_subj_procedure, Craniotomy, "recovery_time_unit"),
+            "craniotomy_size": old_subj_procedure.get("craniotomy_size", None),
+        }
+
+        if not craniotomy_dict['craniotomy_type'] and craniotomy_dict['craniotomy_size']:
+            if '3' in craniotomy_dict['craniotomy_size']:
+                craniotomy_dict['craniotomy_type'] = '3 mm'
+            elif '5' in craniotomy_dict['craniotomy_size']:
+                craniotomy_dict['craniotomy_type'] = '5 mm'
+            
+        drop_unused_fields(craniotomy_dict, 'Craniotomy')
+
+        return construct_new_model(craniotomy_dict, Craniotomy, self.allow_validation_errors)
+
+
+    def construct_ophys_probe(self, probe: dict):
+        """Map legacy OphysProbe model to current version"""
+
+        if probe.get("ophys_probe") is not None:
+            fiber_probe_item = probe.get("ophys_probe")
+            fiber_probe_dict = {
+                "device_type": probe.get("device_type", None),
+                "name": probe.get("name", None),
+                "serial_number": probe.get("serial_number", None),
+                "manufacturer": probe.get("manufacturer", None),
+                "model": probe.get("model", None),
+                "path_to_cad": probe.get("path_to_cad", None),
+                "port_index": probe.get("port_index", None),
+                "additional_set": get_or_default(probe, OphysProbe, "additional_set"),
+                "core_diameter": fiber_probe_item.get("core_diameter", None),
+                "core_diameter_unit": fiber_probe_item.get("core_diameter_unit", None),
+                "numerical_aperture": fiber_probe_item.get("numerical_aperture", None),
+                "ferrule_material": fiber_probe_item.get("ferrule_material", None),
+                "active_length": fiber_probe_item.get("active_length", None),
+                "total_length": fiber_probe_item.get("total_length", None),
+                "length_unit": get_or_default(fiber_probe_item, FiberProbe, "length_unit"),
+            }
+            fiber_probe = construct_new_model(fiber_probe_dict, FiberProbe, self.allow_validation_errors)
+        else:
+            fiber_probe = FiberProbe.model_construct()
```

Comment:
not sure I understand this - why is there an empty fiber probe?

Reply:
`fiber_probe` is a required field in `OphysProbe`, but not all assets have them, so was covering the case where it was missing.